gradle/special-tests.gradle

```diff
@@ -0,0 +1,16 @@
+def special = [
+	'Npm',
+	'Black',
+	'Clang'
+]
+
+tasks.named('test') {
+	useJUnit { excludeCategories special.collect { "com.diffplug.spotless.category.${it}Test" } as String[] }
+}
+
+special.forEach {
+	def category = "com.diffplug.spotless.category.${it}Test"
+	tasks.register("${it}Test", Test) {
+		useJUnit { includeCategories category }
+	}
+}
```

Comment:
As far as I can tell, these new `Test` tasks are not being run on CircleCI yet (although I'd love to be corrected on this!), and regardless I think we'd need Node.js, Python and Clang installed on our CircleCI setup for those tests to work, and I can't see any CircleCI-related changes on this PR, so I'm a little confused. Thoughts?

Reply:
Another good catch!  None of this is being exercised on CircleCI right now, which is a problem.  I did extensive testing on my personal win and mac machines, but this should definitely be tested better.

It's easy to write a cross-platform "machine preparation script" for `black`, but it's a little bit trickier for `clang-format`.  On mac, `brew` gives you `10.0.1` and its hard to get anything else.  On windows, `choco` gives you `10.0.0`, and it's hard to get anything else.  And those will both change over time - the llvm website seems to advertise for `12.0`, which apparently hasn't made it into package managers yet.  If your team is all on the same platform (fairly common), then none of this is a problem: "just run `brew install clang-format`"

It would be easy to make a docker image which has been prepared for specific versions for running `blackTest` and `clangTest`.  However, if a contributor wants to run these tests, they'll have to do that preparation manually, which is a pain.  When I went back and forth between win and mac, I just changed the versions in the tests, which isn't great.

Testing would be a lot easier if #673 and #674 were implemented.  So my testing plan was: when someone has a good idea/PR for improving either of those issues, then that's when we'll run the tests on CI, since it'll be a lot easier at that point.

In the meantime, I'll commit to be being the human-test-runner for PRs which affect these native formatters, and I'm very open to a PR which adds a prepared docker image to offload that burden onto CI :D